executor/diagnostics.go

```diff
@@ -0,0 +1,168 @@
+	sctx.GetSessionVars().InspectionTableCache = map[string]variable.TableSnapshot{}
+	defer func() { sctx.GetSessionVars().InspectionTableCache = nil }()
+
+	failpoint.InjectContext(ctx, "mockMergeMockInspectionTables", func() {
+		// Merge mock snapshots injected from failpoint for test purpose
+		mockTables, ok := ctx.Value("__mockInspectionTables").(map[string]variable.TableSnapshot)
+		if !ok {
+			return
+		}
+		for name, snap := range mockTables {
+			sctx.GetSessionVars().InspectionTableCache[strings.ToLower(name)] = snap
+		}
+	})
+
+	rules := e.extractor.Rules
+	items := e.extractor.Items
+	var finalRows [][]types.Datum
+	for _, r := range inspectionRules {
+		name := r.name()
+		if len(rules) > 0 && !rules.Exist(name) {
+			continue
+		}
+		results := r.inspect(ctx, sctx, items)
+		for _, result := range results {
+			finalRows = append(finalRows, types.MakeDatums(
+				name,
+				result.item,
+				result.value,
+				result.reference,
+				result.severity,
+				result.suggestion,
+			))
+		}
+	}
+	return finalRows, nil
+}
+
+type configInspection struct{}
+
+func (configInspection) name() string {
+	return "config"
+}
+
+func (configInspection) inspect(_ context.Context, sctx sessionctx.Context, filter set.StringSet) []inspectionResult {
+	// check the configuration consistent
+	sql := "select type, `key`, count(*) as c from inspection_schema.cluster_config group by type, `key` having c > 1"
+	rows, _, err := sctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(sql)
+	if err != nil {
+		sctx.GetSessionVars().StmtCtx.AppendWarning(fmt.Errorf("check configuration consistency failed: %v", err))
+	}
+
+	var results []inspectionResult
+	for _, row := range rows {
+		if len(filter) > 0 && !filter.Exist(row.GetString(1)) {
+			continue
+		}
+		results = append(results, inspectionResult{
+			item:       row.GetString(1), // key
+			value:      row.GetString(2), // count
+			reference:  "1",
+			severity:   "P2",
+			suggestion: fmt.Sprintf("select * from cluster_config where type='%s' and `key`='%s'", row.GetString(0), row.GetString(1)),
+		})
+	}
+	return results
+}
+
+type versionInspection struct{}
+
+func (versionInspection) name() string {
+	return "version"
+}
+
+func (versionInspection) inspect(_ context.Context, sctx sessionctx.Context, filter set.StringSet) []inspectionResult {
+	// check the configuration consistent
+	sql := "select type, count(*) as c from inspection_schema.cluster_info group by type, git_hash having c > 1;"
+	rows, _, err := sctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(sql)
+	if err != nil {
+		sctx.GetSessionVars().StmtCtx.AppendWarning(fmt.Errorf("check version consistency failed: %v", err))
+	}
+
+	var results []inspectionResult
+	for _, row := range rows {
+		if len(filter) > 0 && !filter.Exist(row.GetString(0)) {
+			continue
+		}
+		results = append(results, inspectionResult{
+			item:       row.GetString(0), // type
+			value:      row.GetString(1), // count
+			reference:  "1",
```

Comment:
I prefer to print error message, not these fields. Users must query documentation to see what each field means.